overlord/snapstate/snapstate.go

```diff
@@ -497,6 +497,19 @@ func GuardCoreSetupProfilesPhase2(task *state.Task, snapsup *SnapSetup, snapInfo
 		if err != nil {
 			return false, err
 		}
+		// check what snap we use for booting
+		model, err := Model(task.State())
+		if err != nil {
+			return false, err
+		}
+		// if the boot snap does not match the "core" nothing
+		// to do
```

Comment:
Does this assume that when snapd snap is used it restarts itself like on classic? (instead of rebooting the device)